### ISSUE TITLE
Uninstallable via elm-package

### ISSUE THREAD

**Issue by kuliniew** [tapeinosyne/elm-microkanren#5]
Trying to install [the ndr-qef/microkanren.elm package](http://package.elm-lang.org/packages/ndr-qef/microkanren.elm/1.0.1) via `elm-package` fails.

Steps to reproduce:

```
$ mkdir repro
$ cd repro
$ elm --version
0.17.1
$ elm package install --yes
# (base packages install successfully)
$ elm package install --yes ndr-qef/microkanren.elm 1.0.1
Starting downloads...

  ✗ ndr-qef/microkanren.elm 1.0.1
Error: Problem when downloading the ndr-qef/microkanren.elm 1.0.1 code.

$ ls -1 elm-stuff/packages/
elm-lang
ndr-qef-elm-microkanren-ddc30c9
```

Expected behavior:
- `elm package install` succeeds without errors
- `elm-stuff/packages/ndr-qef/microkanren.elm/1.0.1/` exists

I suspect that `elm-package` is getting confused by the GitHub repo being named `ndr-qef/elm-microkanren` but the package being published as `ndr-qef/microkanren.elm`.

**Comment by tapeinosyne** [comment]
It seems that `elm package` fails to follow redirects. Renaming the repository should suffice; let me know if there are still problems.

(Ideally we would change the package name to adhere to the `elm-package` naming convention, but I can find no information on such a process.)

**Comment by kuliniew** [comment]
That does fix the installation problem, but now I'm seeing that the `.` in the package name causes invalid Javascript to be generated:

```
$ elm repl
---- elm-repl 0.17.1 -----------------------------------------------------------
 :help for help, :exit to exit, more at <https://github.com/elm-lang/elm-repl>
--------------------------------------------------------------------------------
> import MicroKanren
/Users/paul/repro/repl-temp-000.js:4063
var _ndr_qef$microkanren.elm$MicroKanren$extend = F3(
                        ^

SyntaxError: Unexpected token .
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:387:25)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:142:18)
    at node.js:939:3

```

I think this is really a bug in the Elm compiler, though, since it's generating a Javascript identifier with an illegal character in it.  I'll file a bug report (assuming there isn't one already).  [Edit: there is: https://github.com/elm-lang/elm-compiler/issues/1475]

**Comment by tapeinosyne** [comment]
As you say, fixing a codegen bug is well outside my power as package maintainer. On the tangent, GitHub repository addresses offer no guarantee of stability, and using them as package identifiers is arguably a mistake.

I'll keep this issue open, but I can offer little help beyond suggesting to use a local fork until the matter is resolved upstream.

**Comment by tapeinosyne** [comment]
I expunged old releases and published a 0.18-compatible version under the guise of a new package. (Old versions, while untagged, can easily be recovered from previous commits.) It's a rather blunt approach, but I have little wish to entertain `elm-package` further.

@kuliniew, you should now be able to install `microkanren` as `tapeinosyne/elm-microkanren` without trouble. Hope it helps.
